benchmark/nixlbench/src/worker/nixl/nixl_worker.cpp

```diff
@@ -1024,31 +1433,78 @@ xferBenchNixlWorker::transfer(size_t block_size,
     return std::variant<xferBenchStats, int>(stats);
 }
 
+#if HAVE_NIXL_DEV_API
+void
+xferBenchNixlWorker::device_poll(size_t block_size, unsigned int skip, unsigned int num_iter) {
+    std::string_view gpu_level = *(xferBenchConfigGpuLevels.find(xferBenchConfig::gdaki_gpu_level));
+    unsigned int total_iter = skip + num_iter;
+
+    // GDAKI mode: Target monitors signal buffer instead of waiting for notifications
+    // Monitor signal buffer if we have one
+    if (!signal_buffers.empty()) {
+        void *signal_addr = reinterpret_cast<void *>(signal_buffers[0].addr);
+        // Wait for warmup iterations to complete
+        uint64_t count = 0;
+        while (count < skip) {
+            count = readNixlGpuSignal(signal_addr, gpu_level.data());
```

Comment:
can we do all polling logic in the kernel? Or need to avoid cudaMalloc/free for every read